cpp/src/arrow/CMakeLists.txt

```diff
@@ -228,48 +219,38 @@ if(ARROW_COMPUTE)
 endif()
 
 if(ARROW_CUDA)
-  # IPC extensions required to build the CUDA library
-  set(ARROW_IPC ON)
   add_subdirectory(gpu)
 endif()
 
-if(ARROW_WITH_BROTLI)
-  add_definitions(-DARROW_WITH_BROTLI)
-  set(ARROW_SRCS util/compression_brotli.cc ${ARROW_SRCS})
-endif()
-
-if(ARROW_WITH_BZ2)
-  add_definitions(-DARROW_WITH_BZ2)
-  set(ARROW_SRCS util/compression_bz2.cc ${ARROW_SRCS})
-endif()
-
-if(ARROW_WITH_LZ4)
-  add_definitions(-DARROW_WITH_LZ4)
-  set(ARROW_SRCS util/compression_lz4.cc ${ARROW_SRCS})
+if(ARROW_DATASET)
+  add_subdirectory(dataset)
 endif()
 
-if(ARROW_WITH_SNAPPY)
-  add_definitions(-DARROW_WITH_SNAPPY)
-  set(ARROW_SRCS util/compression_snappy.cc ${ARROW_SRCS})
-endif()
+if(ARROW_FILESYSTEM)
+  add_subdirectory(filesystem)
 
-if(ARROW_WITH_ZLIB)
-  add_definitions(-DARROW_WITH_ZLIB)
-  set(ARROW_SRCS util/compression_zlib.cc ${ARROW_SRCS})
-endif()
+  set(ARROW_SRCS
```

Comment:
Could you use `list(APPEND ARROW_SRCS ...)`?
https://cmake.org/cmake/help/latest/command/list.html#modification

Reply:
Done.